### ISSUE TITLE
Script Encountered an un-handled exception!

### ISSUE THREAD

**Issue by dkotars** [vilega/O365Troubleshooters#233]
Running the script now seems to generate: Script Encountered an un-handled exception! 

Log shows: "ERROR: Cannot validate argument on parameter 'Session'. The argument is null. Provide a valid value for the argument"

**Comment by batoglu** [comment]
I guess didn't fixed. 

today I faced same issue

System.Management.Automation.ParameterBindingValidationException: Cannot validate argument on parameter 'Session'. The argument is null. Provide a valid value for the argument, and then try running the command

PS version : 5.1.19041.2364

O365Troubleshooters version : 2.0.0.26

**Comment by scalfbenjamin** [comment]
Also seeing the same problem: "ERROR: Cannot validate argument on parameter 'Session'. The argument is null. Provide a valid value for the argument"

Pipes TONS of output to a XML until you ctrl-C

**Comment by afran1488** [comment]
I was able to get get around the null argument error.
#Check if O365Troubleshooters module is installed
if (Get-InstalledModule -Name O365Troubleshooters -ErrorAction SilentlyContinue) {
    # If installed, uninstall it
    Uninstall-Module -Name O365Troubleshooters -Force
}

#Install O365Troubleshooters module from PSGallery repository
Install-Module -Name O365Troubleshooters -Repository PSGallery -Force

#Set TLS 1.2 as the security protocol
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12

#Run the Start-O365Troubleshooters command with elevated execution
Start-O365Troubleshooters -ElevatedExecution $True

**Comment by dkotars** [comment]
Thanks, that does get me to where I need to be.

**Comment by MetUys** [comment]
This still seems to be an issue.
I installed the O365Troubleshooters a day or so ago, but was unable to connect on "task 18", after authenticating the exchange online connection, it would through an unexpected error as described here.
I tried the work around suggested, which worked for the exchange online connection, but then it threw the unexpected error again for the compliance authentication.
I did "rinse and repeat" the work around above and then it allowed me to connect to the compliance connection as well.
(So the bug lives in both processes it seems)

I dont think the work around is a valid way to move forward here, and new users will likely not see this work around as its in a closed bug task. I was about to give up on this tool before I decided to snoop around the closed bugs list in the repo and stumbled onto this. Im glad I have it running now, but it was by chance... and looking at the way this manifest's, I believe it will surface again once my sessions are disconnected.
